includes/multi-currency/FrontendCurrencies.php

```diff
@@ -270,7 +276,16 @@ public function init_order_currency( $arg ) {
 			return $arg;
 		}
 
+		// We remove the filters here becuase 'wc_get_order' triggers the 'wc_get_price_decimal_separator' filter.
+		remove_filter( 'wc_get_price_decimal_separator', [ $this, 'get_price_decimal_separator' ], 900 );
+		remove_filter( 'wc_get_price_decimal_separator', [ $this, 'init_order_currency_from_query_vars' ], 900 );
```

Comment:
Do we have to worry about the priority being the same here too? I don't think so but thought of asking to be sure 🙂

Reply:
yes, we do have to worry about the priority being the same, thank you for catching this! Completely missed it 😅

Reply:
Fixed in https://github.com/Automattic/woocommerce-payments/pull/8387/commits/3f93b1fdc324bfd7a1029265635e63c7afd5f42b